### PR TITLE
InfernalVoidScans: Switch to iken MultiSrc

### LINKED ISSUE #6612: Infernal Void Scan not working
### Source information

Infernal Void Scans 1.4.40

### Source language

English

### Steps to reproduce

1. Open extension
2. No results found
Screenshot_2024-12-15-01-36-17-23_551f7f0ca926862b368845ce7ee433aa

### Expected behavior

1. Open extension 
2. Series shown

### Actual behavior

No series shown

### Mihon/Tachiyomi version

0.17.1

### Android version

Android 14

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to version **[0.15.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] If this is an issue with the app itself, I should be opening an issue in the [app repository](https://github.com/tachiyomiorg/tachiyomi/issues/new/choose).
- [X] I will fill out all of the requested information in this form.

### 

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/keiyoushi/extensions-source/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Creepler13**
Closes:  #6612

Checklist:

- [X] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [X] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [X] Have not changed source names
- [X] Have explicitly kept the `id` if a source's name or language were changed
- [X] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/en/infernalvoidscans/src/eu/kanade/tachiyomi/extension/en/infernalvoidscans/InfernalVoidScans.kt [hunk 1]
```diff
@@ -20,21 +25,26 @@ class InfernalVoidScans : MangaThemesia(
...
-    override val pageSelector = "div#readerarea > p > img"
-
-    override val hasProjectPage = true
+    private val titleCache by lazy {
+        val response = client.newCall(GET("$baseUrl/api/query?perPage=9999", headers)).execute()
+        val data = json.decodeFromString<SearchResponse>(response.body.string())
+        data.posts
+            .filterNot { it.isNovel }
+            .associateBy { it.slug }
+    }
```
Comment (AwkwardPeak7) [inline src/en/infernalvoidscans/src/eu/kanade/tachiyomi/extension/en/infernalvoidscans/InfernalVoidScans.kt]: don't request this again, make the one in multisrc protected so it is accessable here

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]

File: src/en/infernalvoidscans/src/eu/kanade/tachiyomi/extension/en/infernalvoidscans/InfernalVoidScans.kt [hunk 2]
```diff
@@ -20,21 +22,18 @@ class InfernalVoidScans : MangaThemesia(
         }
         .build()
 
+    private val json by injectLazy<Json>()
+
```
Comment (AwkwardPeak7) [inline src/en/infernalvoidscans/src/eu/kanade/tachiyomi/extension/en/infernalvoidscans/InfernalVoidScans.kt]: unused

File: src/en/infernalvoidscans/src/eu/kanade/tachiyomi/extension/en/infernalvoidscans/InfernalVoidScans.kt [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
+import eu.kanade.tachiyomi.multisrc.iken.Iken
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.util.asJsoup
+import kotlinx.serialization.json.Json
+import okhttp3.Response
+import uy.kohesive.injekt.injectLazy
 
-class InfernalVoidScans : MangaThemesia(
+class InfernalVoidScans : Iken(
     "Infernal Void Scans",
```
Comment (AwkwardPeak7) [inline src/en/infernalvoidscans/src/eu/kanade/tachiyomi/extension/en/infernalvoidscans/InfernalVoidScans.kt]: while you are at it, change the name to "Hive Toon" or similar here and in build.gradle file

**Review by AwkwardPeak7 — CHANGES_REQUESTED** [review]
also bump `versionId` since new theme have different url formats

File: src/en/infernalvoidscans/src/eu/kanade/tachiyomi/extension/en/infernalvoidscans/InfernalVoidScans.kt (hunk 3 above)
Comment (Creepler13) [reply src/en/infernalvoidscans/src/eu/kanade/tachiyomi/extension/en/infernalvoidscans/InfernalVoidScans.kt]: should i also change the path/folder names or should that stay the same?

**Review by AwkwardPeak7 — APPROVED** [review]

**Pull request merged**